Data/Tree.hs

```diff
@@ -232,34 +280,111 @@ draw (Node x ts0) = lines x ++ drawSubTrees ts0
 
     shift first other = zipWith (++) (first : repeat other)
 
--- | The elements of a tree in pre-order.
+-- | Returns the elements of a tree in pre-order.
+--
+-- @
+--
+--   a
+--  / \\    => [a,b,c]
+-- b   c
+-- @
+--
+-- ==== __Examples__
+--
+-- > flatten (Node 1 [Node 2 [], Node 3 []]) == [1,2,3]
 flatten :: Tree a -> [a]
 flatten t = squish t []
   where squish (Node x ts) xs = x:Prelude.foldr squish xs ts
 
--- | Lists of nodes at each level of the tree.
+-- | Returns the list of nodes at each level of the tree.
+--
+-- @
+--
+--   a
+--  / \\    => [[a], [b,c]]
+-- b   c
+-- @
+--
+-- ==== __Examples__
+--
+-- > levels (Node 1 [Node 2 [], Node 3 []]) == [[1],[2,3]]
+--
 levels :: Tree a -> [[a]]
 levels t =
     map (map rootLabel) $
         takeWhile (not . null) $
         iterate (concatMap subForest) [t]
 
--- | Catamorphism on trees.
+-- | Fold a tree into a "summary" value in depth-first order.
+--
+-- For each node in the tree, apply @f@ to the @rootLabel@ and the result
+-- of applying @f@ to each @subForent@.
+--
+-- This is also known as the catamorphism on trees.
+--
+-- ==== __Examples__
+--
+-- Sum the values in a tree:
+--
+-- > foldTree (\x xs -> sum (x:xs)) (Node 1 [Node 2 [], Node 3 []]) == 6
+--
+-- Find the maximum value in the tree:
+--
+-- > foldTree (\x xs -> maximum (x:xs)) (Node 1 [Node 2 [], Node 3 []]) == 3
+--
 --
 -- @since 0.5.8
 foldTree :: (a -> [b] -> b) -> Tree a -> b
 foldTree f = go where
     go (Node x ts) = f x (map go ts)
 
--- | Build a tree from a seed value
+-- | Build a (possibly infinite) tree from a seed value in breadth-first order.
+--
+-- @unfoldTree f b@ constructs a tree by starting with the tree
+-- @Node { rootLabel=b, subForest=[] }@ and repeatedly applying @f@ to each
+-- 'rootLabel' value in the tree's leaves to generate its 'subForest'.
+--
+-- For a monadic version see 'unfoldTreeM_BF'.
+--
+-- ==== __Examples__
```

Comment:
There's only one example now.

Reply:
Yeah, I wasn't sure if the other one was useful. I can add it back in if you think it would be.

Reply:
Don't remember what it was, but I suspect one good example should be sufficient.

Reply:
It was a tree that consisted of a chain of 1's `Node 1 [Node 1 [...]]`, not super useful :P